base/twiceprecision.jl

```diff
@@ -193,27 +193,46 @@ roundoff error).  If `step` has an exact rational representation
 where `nb` is the number of trailing zero bits of `step.hi`.  For
 ranges, you can set `nb = ceil(Int, log2(len-1))`.
 """
-struct TwicePrecision{T}
+struct TwicePrecision{T<:Number}
```

Comment:
This breaks the `PhysQuantity` tests, see `test/testhelpers/PhysQuantities.jl` and [their tests](https://github.com/JuliaLang/julia/pull/49616/files#diff-1d6e3ecff651f017ed32b54d0919e13362ac278dd18df1cc8b2feda52e630744L251-R256). I think those are there for ranges of e.g. Unitful quantities, and other non-`Number`s.